lib/active_merchant/billing/gateways/merchant_e_solutions.rb

```diff
@@ -122,6 +125,15 @@ def add_creditcard(post, creditcard, options)
         post[:card_exp_date]  = expdate(creditcard)
       end
 
+      # not a 3dsecure integration - simply passes params to gateway if present,
+      # verified externally
+      def add_3dsecure_params(post, options)
+        post[:xid] = options[:xid] unless options[:xid].blank?
```

Comment:
We're trying to lessen our dependence on ActiveSupport so we're trying to lessen the use of `blank?`.  Example of the approach we're using:

https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L463
https://github.com/activemerchant/active_merchant/blob/master/lib/active_merchant/billing/gateways/authorize_net.rb#L6